wallet/wallet.go

```diff
@@ -3803,19 +3803,90 @@ func (w *Wallet) ResetLockedOutpoints() {
 // LockedOutpoints returns a slice of currently locked outpoints.  This is
 // intended to be used by marshaling the result as a JSON array for
 // listlockunspent RPC results.
-func (w *Wallet) LockedOutpoints() []dcrdtypes.TransactionInput {
+func (w *Wallet) LockedOutpoints(ctx context.Context, accountName string) ([]dcrdtypes.TransactionInput, error) {
 	w.lockedOutpointMu.Lock()
-	locked := make([]dcrdtypes.TransactionInput, len(w.lockedOutpoints))
+	allLocked := make([]wire.OutPoint, len(w.lockedOutpoints))
 	i := 0
 	for op := range w.lockedOutpoints {
-		locked[i] = dcrdtypes.TransactionInput{
-			Txid: op.Hash.String(),
-			Vout: op.Index,
-		}
+		allLocked[i] = op
 		i++
 	}
 	w.lockedOutpointMu.Unlock()
-	return locked
+
+	allAccts := accountName == "" || accountName == "*"
+	acctLocked := make([]dcrdtypes.TransactionInput, 0, len(allLocked))
+	err := walletdb.View(ctx, w.db, func(tx walletdb.ReadTx) error {
+		addrmgrNs := tx.ReadBucket(waddrmgrNamespaceKey)
+		txmgrNs := tx.ReadBucket(wtxmgrNamespaceKey)
+
+		defaultAccountName, err := w.manager.AccountName(
```

Comment:
The handling for the default account appears to be entirely unnecessary.  If the account is specified, only look that one up and compare for equality.  If the account name is "" or "*", then all results should be included.  Neither of these cases rely on falling back to showing results of the default account.